docs/naming_conventions.md

```diff
@@ -0,0 +1,174 @@
+one and stick to it.
+
+Based on observations, the FHIR spec seems to use PascalCase for
+resource types (e.g. `Patient`, `SearchParameter`) and camelCase
+(e.g. `gender`, `birthDate`) for resource attributes
+
+The Kids First model will use [kebab-case](https://stackoverflow.com/questions/11273282/whats-the-name-for-hyphen-separated-case)
+for all resource `id` values. Kebab case is essentially snake case with hyphens
+instead of underscores (e.g. `age-at-event`, `sequencing-experiment`).
+
+
+## Resource File Name
+
+### Conformance Resources
+
+All conformance resource file names should follow this pattern:
+
+**Pattern**
+```
+<resource type hierarchy delimited by underscores>_<resource id>.json
+```
+
+The resource type hierarchy represents the inheritance chain of resource types.
+
+For example if a user created a `StructureDefinition` for `Patient` then
+the resource type hierarchy string would be: `StructureDefinition_Patient`.
+
+For other conformance resources that are not meant to be extended the
+resource type hierarchy will have a depth of 1. For example if a user created
+a `CodeSystem` for ethnicity for US Core, the resource type hierarchy string
+would be something like: `CodeSystem_us-core-ethnicity.json`
+
+### File Name Delimiter
+
+The file name delimiter will be an underscore `_`, since it is not contained
+in the character set for the resource `id`. The delimiter will be used to
+separate the resource type from the resource id in the file name.
+
+**Example StructureDefinition Resource**
+```json
+{
+  "resourceType": "StructureDefinition",
+  "id": "participant",
+  "url": "https://fhir.kids-first.io/StructureDefinition/participant",
+  "name": "participant",
+  "status": "draft",
+  "fhirVersion": "4.0.0",
+  "kind": "resource",
+  "abstract": false,
+  "type": "Patient",
+  "baseDefinition": "https://hl7.org/fhir/StructureDefinition/Patient",
+  "derivation": "constraint",
+  "differential": {
+    "element": [
+      {
+        "id": "Patient.gender",
+        "path": "Patient.gender",
+        "min": 1
+      },
+      {
+        "id": "Patient.birthDate",
+        "path": "Patient.birthDate",
+        "max": "0"
+      }
+    ]
+  }
+}
+```
+
+**File Name Example**
+```
+StructureDefinition_Patient_participant.json
+```
+
+**Additional Examples**
+```
+# Extension type StructureDefinition
+StructureDefinition_Extension_age-at-event.json
+
+# CodeSystem
+CodeSystem_us-core-ethnicity.json
+
+# ValueSet
+ValueSet_us-core-ethnicity.json
+```
+
+### Non-Conformance Resources
+
+All non-conformance resource file names should follow this pattern:
+
+**Pattern**
+```
+<resource type hierarchy>_<resource's profile id>_<resource id>.json
+```
+
+**NOTE:**
+
+- This has to be different than conformance resources since non-conformance
+resources do not have the `baseDefinition` attribute. The rough equivalent
+is the `meta.profile` attribute
+- This naming convention assumes there is one profile in the profile list.
```

Comment:
I think we need a convention that doesn't assume this. I gave an example above.

Reply:
Hmm idk if we should even say we're supporting multiple profiles for the reason we listed [here](https://github.com/kids-first/kf-model-fhir/blob/master/docs/feedback_and_questions/general_fhir_feedback_and_questions.md#structuredefinitions).